### DISCUSSION TITLE
Cordova electron and SQLCipher integration

### DISCUSSION THREAD

**Discussion by marek8623** [reZach/blog#10] [General]
Hi everyone,

I am trying to run my Cordova react app on Mac using cordova-electron [Cordova Electron](https://github.com/apache/cordova-electron) and [journeyapps/node-sqlcipher](https://github.com/journeyapps/node-sqlcipher).

I am using these versions:

"cordova-electron": "^3.1.0",
"@journeyapps/sqlcipher": "^5.3.1"

When I want to access @journeyapps/sqlcipher using require I am getting the following error:

`window.require('@journeyapps/sqlcipher').verbose();`

Error:
_Uncaught (in promise) TypeError: window.require is not a function_

When I use:

"cordova-electron": "^1.1.1",
"@journeyapps/sqlcipher": "^4.1.0"

everything is working okay.

I read this article: [https://github.com/reZach/secure-electron-template/blob/master/docs/newtoelectron.md](https://github.com/reZach/secure-electron-template/blob/master/docs/newtoelectron.md) but I am not sure if this example can help solve my issue.
I also read this article about removing nodeIntegration completely: [https://github.com/electron/electron/issues/23506](https://github.com/electron/electron/issues/23506).

My current browserWindow settings:

```
{
	"browserWindow": {
		"width": 1280,
		"height": 768,
		"webPreferences": {
			"nodeIntegration": true,
		}
	}
}
```

Can someone help me with this issue or has working solution?

Thanks

**Comment by W3LL0** [comment]
Hello Marek , I have the same issue , and I t hink we are not the only one , it starts to be a headache to import module with Electron ... I spoke to several person , and some of them think to stop programming with this framework .... it's so much work to try to adapt the code of a project made with an old version of electron and adapt it to recent versions. I post on stackoverflow yesterday , [see here ](https://stackoverflow.com/questions/74724373/electron-require-is-not-defined-electron-version-12-nodeintegration-chan) ........ maybe someone will help us :)
